.github/workflows/release.yaml

```diff
@@ -62,22 +57,28 @@ jobs:
           git branch --set-upstream-to=origin/master
           git pull
       - name: Update Release Version ${{steps.metadata.outputs.release-version}}
-        run: mvn versions:set -DnewVersion=${{steps.metadata.outputs.release-version}} -DgenerateBackupPoms=false -DprocessAllModules=true
+        run: cd registry && mvn versions:set -DnewVersion=${{steps.metadata.outputs.release-version}} -DgenerateBackupPoms=false -DprocessAllModules=true
       - name: Build All Variants
-        run: mvn clean install -Pprod -Pjpa -Pinfinispan -Pkafka -Pstreams -Pasyncmem -DskipTests
+        run: cd registry && mvn clean install -Pprod -Pjpa -Pinfinispan -Pkafka -Pstreams -Pasyncmem -DskipTests
       - name: Commit Release Version Change
         run: |
+          cd registry
           git add .
           git commit -m "Automated update to Release Version:: ${{steps.metadata.outputs.release-version}}"
           git push
-      - name: Publish to Maven Central
-        uses: samuelmeuli/action-maven-publish@v1
-        with:
-          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
-          gpg_passphrase: ${{ secrets.GPG_PASSPHRASE }}
-          nexus_username: ${{ secrets.OSSRH_USERNAME }}
-          nexus_password: ${{ secrets.OSSRH_TOKEN }}
-          maven_goals_phases: deploy --batch-mode -Pprod -Pjpa -Pinfinispan -Pkafka -Pstreams -Pasyncmem -DskipTests -Prelease --settings /home/runner/.m2/settings.xml -DskipTests
+      - name: Import GPG Key
+        uses: crazy-max/ghaction-import-gpg@v1
+        env:
+          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+          PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }} 
+      - name: Maven Deploy
+        run: |
+          cd registry
+          # Retry 3 times before the steps actually fails
+          (echo "===== Maven Deploy Attempt: 1 ====" && mvn deploy --batch-mode -Pprod -Pjpa -Pinfinispan -Pkafka -Pstreams -Pasyncmem -DskipTests -Prelease --settings /home/runner/.m2/settings.xml -DskipTests) || \
+          (echo "===== Maven Deploy Attempt: 2 ====" && mvn deploy --batch-mode -Pprod -Pjpa -Pinfinispan -Pkafka -Pstreams -Pasyncmem -DskipTests -Prelease --settings /home/runner/.m2/settings.xml -DskipTests) || \
+          (echo "===== Maven Deploy Attempt: 3 ====" && mvn deploy --batch-mode -Pprod -Pjpa -Pinfinispan -Pkafka -Pstreams -Pasyncmem -DskipTests -Prelease --settings /home/runner/.m2/settings.xml -DskipTests) || \
+          (echo "==== Maven Deploy Step Failed ====" && exit 1)
```

Comment:
Can you explain why this works?  I know `&&` can be used to string together multiple commands - but I'm unfamiliar with the gritty details.

E.g. does this stop trying if the first attempt succeeds?

@jsenko and @carlesarnal - I would appreciate your thoughts on this.  I'm wondering if a script might be better here.

Reply:
@riprasad I realized my confusion here is because you used the `&&` operator between the `echo` and `mvn` commands and then you used the `||` operator after that.  I didn't catch that on first reading and thought/assumed you were using `&&` between each command (which is what I've seen often in e.g. Dockerfiles).  So I was not understanding the code.  But now I do and it makes sense.  The `||` operator isn't something I've used before but looking at the docs for it it makes sense.

Reply:
@EricWittmann Yes, correct! 
Sorry for the delayed response though. I always miss mails from Github. I have a filter in place now to have all Github mails in one folder for better visibility and tracking.

Reply:
Thank you, @riprasad - I do the same thing (gmail filter for GH emails).